tools/Resizer/resizer.py

```diff
@@ -0,0 +1,308 @@
+                                    + self.matrix[i][j][rgb]*(1-col_color_proportion)
+                                )
+                            # we have no sample pixels literally anywhere
+                            else:
+                                value = self.matrix[i][j][rgb]
+
+                            # append calculated color to the last pixel
+                            resized_result[-1][-1].append(value)
+        self.file_size = self.file_size\
+            + 3*(self.width*self.height)*(coef**2 - 1)
+        self.width = int(self.width * coef)
+        self.height = int(self.height * coef)
+        self.matrix = resized_result
+
+    def print_info(self):
+        ''' Prints headers into console '''
+        print("\n\tHeader")
+        print("Signature\t", self.signature)
+        print("FileSize\t", self.file_size, "bits")
+        print("reserved\t", self.reserved)
+        print("DataOffset\t", self.offset)
+
+        print("\n\tInfoHeader")
+        print("Size\t\t", self.size)
+        print("Width\t\t", self.width)
+        print("Height\t\t", self.height)
+        print("Bits Per Pixel\t", self.bits_per_pixel)
+        print("Compression\t", self.compression)
+        print("ImageSize\t", self.image_size)
+        print("XpixelsPerM\t", self.x_pixels_per_m)
+        print("YpixelsPerM\t", self.y_pixels_per_m)
+        print("Colors Used\t", self.colors_used)
+        print("Impotant Colors\t", self.important_colors)
+
+    def print_color_table(self):
+        ''' Prints colortable into console '''
+        print("\n\tColorTable")
+        for i in range(int((self.offset-54)/4)):
+            print(self.color_table.read(4))
+
+    def print_image(self):
+        ''' Prints every pixel`s bytes into console '''
+        print("\n\tImage")
+        print(self.matrix)
+
+    def write_to_file(self, destination):
+        ''' Writes updated image to a new file '''
+        # Signature (0-1)
+        destination.seek(0, 0)
+        destination.write(self.signature)
+        # FileSize (2-5)
+        write_bytes_from_int(2, 4, self.file_size, destination)
+        # reserved (6-9)
+        write_bytes_from_int(6, 4, self.reserved, destination)
+        # DataOffset (10-13)
+        write_bytes_from_int(10, 4, self.offset, destination)
+        # Size (of info header) (14-17), defaults to 40
+        write_bytes_from_int(14, 4, 40, destination)
+        # Width (18-21)
+        write_bytes_from_int(18, 4, self.width, destination)
+        # Height (22-25)
+        write_bytes_from_int(22, 4, self.height, destination)
+        # Planes (26-27)
+        write_bytes_from_int(26, 2, self.planes, destination)
+        # Bits Per Pixel (28-29)
+        write_bytes_from_int(28, 2, self.bits_per_pixel, destination)
+        # Compression (30-33)
+        write_bytes_from_int(30, 4, self.compression, destination)
+        # (compressed) ImageSize (34-37), defaults to 0
+        write_bytes_from_int(34, 4, 0, destination)
```

Comment:
Magic numbers here and everywhere
![AmbitiousPracticalJabiru-small](https://user-images.githubusercontent.com/4405297/120245476-fa3a0280-c275-11eb-853d-52fea1dda899.gif)

Reply:
This is not a magic numbers, they are used in the bmp file format, which are not going to change soon ._.)
Any way to remove them will bring just move confusion as what they are reffering to

Reply:
http://www.ece.ualberta.ca/~elliott/ee552/studentAppNotes/2003_w/misc/bmp_file_format/bmp_file_format.htm

Reply:
the relevant fields are described above the function that writes to them